src/execution/exec/execution_context.cpp

```diff
@@ -0,0 +1,26 @@
+#include "execution/exec/execution_context.h"
+#include "execution/sql/value.h"
+
+namespace terrier::exec {
+
+ExecutionContext::StringAllocator::StringAllocator() : region_("") {}
+
+ExecutionContext::StringAllocator::~StringAllocator() = default;
+
+char *ExecutionContext::StringAllocator::Allocate(std::size_t size) {
+  return reinterpret_cast<char *>(region_.Allocate(size));
+}
+
+void ExecutionContext::StringAllocator::Deallocate(UNUSED char *str) {
+  // No-op. Bulk de-allocated upon destruction.
```

Comment:
This should definitely be in the header since it is a no-op.  We're likely forcing an unnecessary function call by not allowing this one to be inlined.